### ISSUE TITLE
Flash of odd colors on explosions, energy powerups, etc.

### ISSUE THREAD

**Issue by roncli** [CDarrow/DXX-Retro#95]
Occasionally you will see a flash of odd color on an explosion or an energy powerup:

![image](https://user-images.githubusercontent.com/498929/35315922-c8bd1b0a-0083-11e8-9428-08d0c8f3652f.png)

![image](https://user-images.githubusercontent.com/498929/35315990-0fcc6c80-0084-11e8-870a-73a02ae04ea1.png)

I have traced this down to code added to the end of ogl.c ogl_loadbmtexture_f in 619b789.  Commenting out that entire code block fixes the problem, but probably screws up custom colors in some way.  I have no idea what this code is doing, and thus no idea on how to fix it.

**Comment by ryusei117** [comment]
I tested this bug and discovered two details that might be useful. It's present in both D1 and D2 - but the light-blue flashes that occur on energy powerups seem to be present only in D2. Also, I could only reproduce the bug in a multiplayer game with "Alternate ship colors" enabled. With the setting disabled or in a single-player game, the flashes didn't occur. Hope that helps!

**Comment by roncli** [comment]
Correct, this was co-op for me, and having them on energy powerups is only occasional.  I got them to happen in D1 level 5, for instance.  Never on single player.

It's definitely that block of code, commenting it out fixes the issue, but probably breaks something else.

**Comment by ryusei117** [comment]
Oh yeah, you're right about D1 level 5 ... So it's both games, then, and it can be level-dependent. Weird. :-S

**Comment by roncli** [comment]
I figured out the problem (and actually read the code to see what it's doing!).  Basically, there is a file-level variable `filter_blueship_wing` that controls when to apply that filter.  But, it's file-level and apparently some of this code runs in parallel, meaning it will filter things it shouldn't.  So to resolve that, I removed the file-level variable and just passed a filter variable into the calling functions.  Worked like a charm.  roncli/DXX-Retro@0b0297b is the commit with the fix.
